src/reducers/toolbarOverlay.js

```diff
@@ -0,0 +1,23 @@
+export const showOverlay = (state, { id }) => ({
+  toolbarOverlay: {
+    showOverlay: true,
+    shortcutId: id
+  }
```

Comment:
This can be accomplished more straightforwardly with a single id. When `showOverlay` is true, id is always truthy, and when `showOverlay` is false, `shortcutId` is `null`, therefore the same information is available if you just store the `id`. It does not need to be a more complex object.

Reply:
I wouldn't be able to maintain some of the functionality, for example, I update the shortcutId on hover, that means the overlay would start showing every time someone hovers over any of the shortcuts

Reply:
Thank you for explaining the purpose. That makes more sense to me now. However I still see it as a suboptimal solution. There is no need to update the `shortcutId` on hover when the overlay is not being shown. If you add the proper condition to onHover and only call `overlayReveal` when the overlay is already revealed, there will be no need for the separate state variable. Thus your state is simpler and you are not calling the hover handler when it is unnecessary.

Reply:
Got it